spec/lib/miq_preloader_spec.rb

```diff
@@ -55,45 +56,67 @@
       vms = Vm.where(:ems_id => ems.id)
       vms2 = vms.order(:id).to_a # preloaded vms to be used in tests
 
-      # there is a query for every ems
-      expect { preload(ems, :vms, vms) }.to make_database_queries(:count => 1)
+      # there is now two queries - rails 7, why?
+      expect { preload(ems, :vms, vms) }.to make_database_queries(:count => 2)
+
       expect { preload(ems, :vms, vms) }.not_to make_database_queries
       expect { expect(ems.vms).to match_array(vms2) }.not_to make_database_queries
       # vms does not get loaded, so it is not preloaded
-      expect { expect(vms.first.ext_management_system).to eq(ems) }.to make_database_queries(:count => 2)
+      expect { expect(vms.first.ext_management_system).to eq(ems) }.to make_database_queries(:count => 1)
     end
 
     it "preloads with a loaded relation (records is a relation)" do
       ems
       emses = ExtManagementSystem.all.load
       vms   = Vm.where(:ems_id => emses.select(:id)).load
-      expect { preload(emses, :vms, vms) }.not_to make_database_queries
+
+      # there is now two queries - rails 7, why?
+      expect { preload(emses, :vms, vms) }.to make_database_queries(:count => 1)
 
       expect { preload(emses, :vms, vms) }.not_to make_database_queries
       expect { expect(emses.first.vms.size).to eq(2) }.not_to make_database_queries
-      expect { expect(vms.first.ext_management_system).to eq(ems) }.not_to make_database_queries
+
+      # This PR seems to indicate only the singular associations are preloaded after the rails 7 refactor
```

Comment:
Very minor, but can you mark this with a TODO tag?

```suggestion
      # TODO: This PR seems to indicate only the singular associations are preloaded after the rails 7 refactor
```

Reply:
Perhaps the other Rails 7 things with the database query counts should also be marked as TODO ?

Reply:
Thanks, that's a good idea. Keenan and I were discussing it yesterday and they're the only thing we've left to look at in this PR.  We're trying a quick dive into them to see if we can determine why they need to change and we'll either mark them TODO as you suggest, pending (for failed assertions that we previously thought should still work... see line 81 below), or some combination.